### PR TITLE
Colors persist when refreshing the log

### PR DESCRIPTION
**Pull request by likeawindrammer**

### CODE REVIEW

File: include/chatlogpiece.h [hunk 1]
```diff
@@ -10,24 +10,25 @@ class chatlogpiece {
 public:
   chatlogpiece();
   chatlogpiece(QString p_name, QString p_showname, QString p_message,
-               bool p_song);
+               bool p_song,int color);
   chatlogpiece(QString p_name, QString p_showname, QString p_message,
-               bool p_song, QDateTime p_datetime);
+               bool p_song, int color,QDateTime p_datetime);
```
Comment (Crystalwarrior) [inline include/chatlogpiece.h:15]: Tiny nitpick, needs a space after the comma

File: src/chatlogpiece.cpp [hunk 2]
```diff
@@ -5,27 +5,31 @@ chatlogpiece::chatlogpiece()
...
   message = p_message;
   p_is_song = p_song;
+  color = p_color;
   datetime = QDateTime::currentDateTime().toUTC();
 }
 
 chatlogpiece::chatlogpiece(QString p_name, QString p_showname,
-                           QString p_message, bool p_song, QDateTime p_datetime)
+                           QString p_message, bool p_song, int p_color,
+                           QDateTime p_datetime)
```
Comment (Crystalwarrior) [inline src/chatlogpiece.cpp:26]: p_color and p_datetime were re-ordered, needs to be double-checked that it doesn't affect anything

File: src/courtroom.cpp [hunk 3]
```diff
@@ -1785,7 +1785,7 @@ void Courtroom::handle_chatmessage(QStringList *p_contents)
     f_charname = ao_app->get_showname(char_list.at(f_char_id).name);
 
   chatlogpiece *temp =
-      new chatlogpiece(f_charname, f_showname, m_chatmessage[MESSAGE], false);
+      new chatlogpiece(f_charname, f_showname, m_chatmessage[MESSAGE], false,m_chatmessage[TEXT_COLOR].toInt());
```
Comment (Crystalwarrior) [inline src/courtroom.cpp:1788]: Add a comma after the space

File: src/courtroom.cpp [hunk 4]
```diff
@@ -1794,7 +1794,7 @@ void Courtroom::handle_chatmessage(QStringList *p_contents)
     ic_chatlog_history.removeFirst();
   }
 
-  append_ic_text(m_chatmessage[MESSAGE], f_showname);
+  append_ic_text(m_chatmessage[MESSAGE], f_showname,"",m_chatmessage[TEXT_COLOR].toInt());
```
Comment (Crystalwarrior) [inline src/courtroom.cpp:1797]: comma spaces pls

File: src/courtroom.cpp [hunk 5]
```diff
@@ -2480,7 +2480,7 @@ QString Courtroom::filter_ic_text(QString p_text, bool html, int target_pos,
   return p_text_escaped;
 }
 
-void Courtroom::append_ic_text(QString p_text, QString p_name, QString p_action)
+void Courtroom::append_ic_text(QString p_text, QString p_name, QString p_action,int color)
```
Comment (Crystalwarrior) [inline src/courtroom.cpp:2483]: comma space durr

File: src/courtroom.cpp [hunk 6]
```diff
@@ -4507,14 +4507,14 @@ void Courtroom::on_showname_enable_clicked()
         append_ic_text(item.get_message(), item.get_showname(),
                        tr("has played a song"));
       else
-        append_ic_text(item.get_message(), item.get_showname());
+        append_ic_text(item.get_message(), item.get_showname(),"",item.get_chat_color());
```
Comment (Crystalwarrior) [inline src/courtroom.cpp:4510]: space!!!!!!

File: src/courtroom.cpp [hunk 7]
```diff
@@ -4507,14 +4507,14 @@ void Courtroom::on_showname_enable_clicked()
...
     }
     else {
       if (item.is_song())
         append_ic_text(item.get_message(), item.get_name(),
                        tr("has played a song"));
       else
-        append_ic_text(item.get_message(), item.get_name());
+        append_ic_text(item.get_message(), item.get_name(),"",item.get_chat_color());
```
Comment (Crystalwarrior) [inline src/courtroom.cpp:4517]: !!!!!!!!

File: src/chatlogpiece.cpp (hunk 2 above)
Comment (likeawindrammer) [reply src/chatlogpiece.cpp:26]: uh, argument order on declaration and definition is consistent.

**Pull request merged**